llvm/test/Instrumentation/MemorySanitizer/SPIRV/instrument_global_address_space.ll

```diff
@@ -0,0 +1,140 @@
+; RUN: opt < %s -passes=msan -msan-instrumentation-with-call-threshold=0 -msan-eager-checks=1 -S | FileCheck %s
+
+; ModuleID = 'check_call.cpp'
+source_filename = "check_call.cpp"
+target datalayout = "e-i64:64-v16:16-v24:32-v32:32-v48:64-v96:128-v192:256-v256:256-v512:512-v1024:1024-n8:16:32:64-G1"
+target triple = "spir64-unknown-unknown"
+
+$_ZTSZZ4mainENKUlRN4sycl3_V17handlerEE_clES2_E8MyKernel = comdat any
+
+; CHECK: @__MsanKernelMetadata
+
+; Function Attrs: mustprogress norecurse nounwind sanitize_memory uwtable
+define weak_odr dso_local spir_kernel void @_ZTSZZ4mainENKUlRN4sycl3_V17handlerEE_clES2_E8MyKernel(ptr addrspace(1) noundef align 4 %_arg_array) local_unnamed_addr #0 comdat !srcloc !85 !kernel_arg_buffer_location !86 !sycl_fixed_targets !87 {
+; CHECK-LABEL: @_ZTSZZ4mainENKUlRN4sycl3_V17handlerEE_clES2_E8MyKernel
+entry:
+; CHECK-NOT: @__msan_param_tls
+  %0 = load i32, ptr addrspace(1) %_arg_array, align 4, !tbaa !88
+  %arrayidx3.i = getelementptr inbounds i8, ptr addrspace(1) %_arg_array, i64 4
+; CHECK: @__msan_get_shadow
+  %1 = load i32, ptr addrspace(1) %arrayidx3.i, align 4, !tbaa !88
+  %conv.i = sext i32 %1 to i64
+  %call.i = tail call spir_func noundef i64 @_Z3fooix(i32 noundef %0, i64 noundef %conv.i) #2
+  %conv4.i = trunc i64 %call.i to i32
+  store i32 %conv4.i, ptr addrspace(1) %_arg_array, align 4, !tbaa !88
+  ret void
+}
+
+; Function Attrs: mustprogress noinline norecurse nounwind sanitize_memory uwtable
+define linkonce_odr dso_local spir_func noundef i64 @_Z3fooix(i32 noundef %data1, i64 noundef %data2) local_unnamed_addr #1 !srcloc !92 {
+; CHECK-LABEL: @_Z3fooix
+entry:
+  %conv = sext i32 %data1 to i64
+  %add = add nsw i64 %data2, %conv
+  ret i64 %add
+}
+
+; CHECK: "sycl-device-global-size"="16" "sycl-device-image-scope" "sycl-host-access"="0" "sycl-unique-id"="_Z20__MsanKernelMetadata"
```

Comment:
Strictly speaking, this line does not check that those attributes are attached to the right LLVM IR entity